plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/GenericRewrite.java

```diff
@@ -37,12 +40,20 @@ public class GenericRewrite
 
     private final ExpressionPattern expressionPattern;
     private final String rewritePattern;
+    // skip in case of any Case insensitive column presents in pattern
+    private final boolean checkCase;
```

Comment:
I'm not sure if `GenericRewrite` should handle a feature specific to varchar columns ?

cc: @hashhar / @wendigo / @findepi your thoughts ?